public/app/features/alerting/unified/components/rule-editor/labels/LabelsFieldInForm.tsx

```diff
@@ -0,0 +1,61 @@
+import { css } from '@emotion/css';
+import React from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Button, Icon, Stack, Text, useStyles2 } from '@grafana/ui';
+
+import { RuleFormValues } from '../../../types/rule-form';
+import { NeedHelpInfo } from '../NeedHelpInfo';
+
+import { LabelsInRule } from './LabelsField';
+
+interface LabelsFieldInFormProps {
+  onEditClick: () => void;
+}
+export function LabelsFieldInForm({ onEditClick }: LabelsFieldInFormProps) {
+  const styles = useStyles2(getStyles);
+  const { watch } = useFormContext<RuleFormValues>();
+  const labels = watch('labels');
+  const hasLabels = Object.keys(labels).length > 0 && labels.some((label) => label.key || label.value);
+
+  return (
+    <div>
+      <Stack direction="column" gap={1}>
+        <Text element="h5">Labels</Text>
+        <Stack direction={'row'} gap={1}>
+          <Text variant="bodySmall" color="secondary">
+            Add labels to your rule for searching, silencing, or routing to a notification policy.
+          </Text>
+          <NeedHelpInfo
+            contentText="The dropdown only displays labels that you have previously used for alerts.
+              Select a label from the options below or type in a new one."
+            title="Labels"
+          />
+        </Stack>
+      </Stack>
+      <div className={styles.labelsContainer}></div>
+      <Stack direction="row" gap={1} alignItems="center">
+        <LabelsInRule />
+        {hasLabels ? (
+          <Icon name={'pen'} onClick={onEditClick} className={styles.editIcon} />
```

Comment:
I think it'd be good to have the edit case show as a button as well, as I've found it easy to miss the icons for editing (e.g. the groups one as well)

Reply:
We previously decided to use this pattern with selected dashboard and panel. I think we should be consistent with the rest of the UI. Maybe we could update styles to make both icons more easy to find?

Reply:
Nice! Yes I think that's easier to spot compared to the icon before :D